### PR TITLE
Initial steps for implementing CaptureOne method in Oscilloscope UI

### LINKED ISSUE #200: Implement captureOne method in Oscilloscope UI.
**Actual Behaviour**

Since we are done with basic Oscilloscope UI, functionality needs to be added to UI. Currently, Oscilloscope UI is not functional.

**Expected Behaviour**

Fetch an oscilloscope trace from the specified input channel and plot it into the graph.

**Steps to reproduce it**

Open Oscilloscope UI and under Channel Parameters select any one input channel.

**LogCat for the issue**

none

**Screenshots of the issue**

none.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by akarshan96**
Fixes #200 and #212 

Changes: 
- Implemented captureOne method
- Prevented Oscilloscope Activity to sleep while running.

Things still left to be done
- Remove read errors
- Determine no. of samples and time-gaps 
- speed up the process

Screenshots for the change: 

screenshot_2017-06-22-19-47-07-822_org fossasia pslab

### CODE REVIEW

File: app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java [hunk 1]
```diff
@@ -629,7 +629,7 @@ public void close() {
         return null;
     }
 
-    private void captureFullSpeedInitialize(String channel, int samples, double timeGap, List<String> args, Integer interval) {
+    public void captureFullSpeedInitialize(String channel, int samples, double timeGap, List<String> args, Integer interval) {
```
Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: would be private. Not used directly.

File: app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java [hunk 2]
```diff
@@ -667,19 +667,21 @@ else if (args.contains("FIRE_PULSES")) {
         }
     }
 
-    public Map<String, double[]> captureFullSpeed(String channel, int samples, double timeGap, List<String> args, Integer interval) {
+    public HashMap<String, double[]> captureFullSpeed(String channel, int samples, double timeGap, List<String> args, Integer interval) {
         /*
         * Blocking call that fetches oscilloscope traces from a single oscilloscope channel at a maximum speed of 2MSPS
         */
```
Comment (viveksb007) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: commented the main functionality why?

Comment (akarshan96) [inline app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java]: My bad, I initially moved captureFullSpeedInitialize in the onPreExecute method of AsyncTask. Then I moved it in the background thread with captureFullSpeed. Let me do the necessary changes.

**Comment by jithinbp** [comment]
@akarshan96 , I'm slightly confused about the use of captureFullspeed . Try to use capture_traces or its derivatives . capture_traces supports simultaneous acquisition of traces from up to 4 channels.
 captureFullspeed uses a DMA channel to record data from a single channel only, but since it can push the sampling rate to 2MSPS, it can be applied in certain experiments .
The data transfer speed issue does not concern this UI, it pertains to the communication thread, and can be dealt with in a separate issue

**Comment by akarshan96** [comment]
Sure @jithinbp, I will try using capture_traces.
Thanks

**Comment by viveksb007** [comment]
@akarshan96 pulled your branch and tried a few changes. This is what I have got currently. As we thought it was timeout due to getAcknowledgement only.
[New Screencast link](https://goo.gl/photos/eEHqMGJmCKcm66K29)
@jithinbp @asitava1998 @CloudyPadmal have a look too.

**Comment by CloudyPadmal** [comment]
@akarshan96 , @viveksb007 when the no. of samples and timegaps are optimized, will it give user a real time screen updates without lags? Or will it be more laggy?

**Comment by viveksb007** [comment]
@akarshan96 please pull changes from this branch and resolve conflicts.
Thanks

**Pull request merged**